rpc/backend/account_info_test.go

```diff
@@ -0,0 +1,357 @@
+				RegisterBlockError(client, bn.Int64())
+			},
+			false,
+			nil,
+		},
+		{
+			"fail - query client failed to get balance",
+			tests.GenerateAddress(),
+			rpctypes.BlockNumberOrHash{BlockNumber: &blockNr},
+			func(bn rpctypes.BlockNumber, addr common.Address) {
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlock(client, bn.Int64(), nil)
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterBalanceError(queryClient, addr, bn.Int64())
+			},
+			false,
+			nil,
+		},
+		{
+			"fail - invalid balance",
+			tests.GenerateAddress(),
+			rpctypes.BlockNumberOrHash{BlockNumber: &blockNr},
+			func(bn rpctypes.BlockNumber, addr common.Address) {
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlock(client, bn.Int64(), nil)
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterBalanceInvalid(queryClient, addr, bn.Int64())
+			},
+			false,
+			nil,
+		},
+		{
+			"fail - pruned node state",
+			tests.GenerateAddress(),
+			rpctypes.BlockNumberOrHash{BlockNumber: &blockNr},
+			func(bn rpctypes.BlockNumber, addr common.Address) {
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlock(client, bn.Int64(), nil)
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterBalanceNegative(queryClient, addr, bn.Int64())
+			},
+			false,
+			nil,
+		},
+		{
+			"pass",
+			tests.GenerateAddress(),
+			rpctypes.BlockNumberOrHash{BlockNumber: &blockNr},
+			func(bn rpctypes.BlockNumber, addr common.Address) {
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlock(client, bn.Int64(), nil)
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterBalance(queryClient, addr, bn.Int64())
+			},
+			true,
+			(*hexutil.Big)(big.NewInt(1)),
+		},
+	}
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.name), func() {
+			suite.SetupTest()
+
+			// avoid nil pointer reference
+			if tc.blockNrOrHash.BlockNumber != nil {
+				tc.registerMock(*tc.blockNrOrHash.BlockNumber, tc.addr)
+			}
+
+			balance, err := suite.backend.GetBalance(tc.addr, tc.blockNrOrHash)
+			if tc.expPass {
+				suite.Require().NoError(err)
+				suite.Require().Equal(tc.expBalance, balance)
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *BackendTestSuite) TestGetTransactionCount() {
+	testCases := []struct {
+		name         string
+		accExists    bool
+		blockNum     rpctypes.BlockNumber
+		registerMock func(common.Address, rpctypes.BlockNumber)
+		expPass      bool
+		expTxCount   hexutil.Uint64
+	}{
+		{
+			"pass - account doesn't exist",
+			false,
+			rpctypes.NewBlockNumber(big.NewInt(1)),
+			func(addr common.Address, bn rpctypes.BlockNumber) {},
+			true,
+			hexutil.Uint64(0),
+		},
+		// TODO Check how to mock RegisterABCIQueryWithOptions correctly
```

Comment:
Maybe this test could be done using `SendTransaction`? or it might be more appropriate to test it in the Nix integration tests as `eth_getTransactionCount`.